pkg/controller/worker/machines.go

```diff
@@ -204,6 +205,21 @@ func (w *workerDelegate) generateMachineConfig() error {
 			machineClassSpec["labels"] = map[string]string{corev1.LabelZoneFailureDomain: zone}
 			machineClassSpec["secret"].(map[string]interface{})["labels"] = map[string]string{v1beta1constants.GardenerPurpose: genericworkeractuator.GardenPurposeMachineClass}
 
+			// if the shoot K8s version is greater than or equal to 1.22 then
+			// enable source destination check on AWS instances by default
+			shootK8sVersion, err := semver.NewVersion(w.cluster.Shoot.Spec.Kubernetes.Version)
+			if err != nil {
+				return errors.Wrapf(err, "Invalid version string '%s'", w.cluster.Shoot.Spec.Kubernetes.Version)
+			}
+			supportedK8sVersion, err := semver.NewVersion("1.22")
+			if err != nil {
+				return errors.Wrapf(err, "Invalid version string 1.22")
+			}
+
+			if shootK8sVersion.Compare(supportedK8sVersion) >= 0 {
+				machineClassSpec["srcAndDstChecksEnabled"] = false
+			}
```

Comment:
FYI: You could use https://github.com/gardener/gardener/blob/6cdb8dc690c2464558c228105852eb3ac7c2c05a/pkg/utils/version/version.go#L28-L35 which would make this a bit shorter, I guess :)

Reply:
Thank you, @rfranzke! Version utils is pretty handy.